src/App/Fossa/VSIDeps.hs

```diff
@@ -57,21 +66,37 @@ analyzeVSIDeps dir projectRevision filters skipResolving = do
           $ rules
 
   resolvedUserDeps <- resolveUserDefined userDeps
-  directSrcUnits <- traverse ruleToSourceUnit directRules
+  directSrcUnits <- traverse (ruleToSourceUnit skipResolving) directRules
 
   -- These deps have to get up to the backend somehow on a 'SourceUnit's 'additionalData'.
   -- This generates an empty-graph source unit and puts the userdeps on it.
   let userDepSrcUnits = toSourceUnit (toProject dir mempty) resolvedUserDeps
 
   pure . Just $ userDepSrcUnits : directSrcUnits
-  where
-    ruleToSourceUnit :: (Has Diagnostics sig m, Has FossaApiClient sig m) => VSI.VsiRule -> m (SourceUnit)
-    ruleToSourceUnit VSI.VsiRule{..} = do
-      resolvedGraph <- resolveGraph [vsiRuleLocator] skipResolving
-      dependencies <- fromEither $ Graphing.gtraverse VSI.toDependency resolvedGraph
-      case parseAbsDir (toString vsiRulePath) of
-        Just ruleDir -> pure $ toSourceUnit (toProject ruleDir dependencies) Nothing
-        Nothing -> fatal $ "Could not parse rule path: " <> show vsiRulePath
+
+#ifdef mingw32_HOST_OS
+-- The path we get back from Core is a Unix-style path, so we need to turn it into a Windows-style path by
+-- prepending a drive (e.g. "C:\") to it and running `normalise` on it.
+-- For example, if we get a path like "/Framework/Common/fjlic/private/WINx64/", we want to end up with "C:\Framework\Common\fjlic\private\WINx64\"
+ruleToSourceUnit :: (Has Diagnostics sig m, Has FossaApiClient sig m, Has ReadFS sig m) => VSI.SkipResolution -> VSI.VsiRule -> m (SourceUnit)
+ruleToSourceUnit skipResolving VSI.VsiRule{..} = do
```

Comment:
Note that the VSI results _appear_ to be absolute paths, but are actually relative to the scan directory (Sherlock considers `/` the "root of the scan", not the "root of the drive").

Does this account for this fact? Specifically, if we scan a project at `C:\Users\me\Documents\projects\foo`, and get back a path like `/bar/baz`, the full path should be `C:\Users\me\Documents\projects\foo\bar\baz`, not `C:\bar\baz`.